main.py

```diff
@@ -551,17 +569,23 @@ async def main(
         filter_tags=filter_tags,
         filter_include_untagged=filter_include_untagged,
         dry_run=dry_run,
+        use_github_token=use_github_token,
     )
     async with AsyncClient(
         headers={'accept': 'application/vnd.github.v3+json', 'Authorization': f'Bearer {token}'}
     ) as client:
         # Get all packages from the user or orgs account
-        all_packages = await GithubAPI.list_packages(
-            account_type=inputs.account_type, org_name=inputs.org_name, http_client=client
-        )
+        if inputs.use_github_token:
+            packages_to_delete_from = set(inputs.image_names)
```

Comment:
The parsing of `image_names` already disallows multiple values, so perhaps we don't need this?
```suggestion
        if inputs.use_github_token:
            packages_to_delete_from = inputs.image_names
```

Reply:
Ah yeah, I converted to `set` because `filter_image_names` was returning a `set`, and this seemed like the path to least resistance. `mypy` was unhappy that `package_to_delete_from` was assigned a `list` while the existing block was a `set`:
```
[Incompatible types in assignment (expression has type "set[str]", variable has type "list[str]")](error: Incompatible types in assignment (expression has type "set[str]", variable has type "list[str]")  [assignment])
```

Alternatively, we might be able to convert the inputs to `set`s, but that would probably be better in a separate, standalone PR cause I foresee a bunch of changes in main and its test to satisfy the type change.

Reply:
As long as there's a reason, it's fine :+1: The code here should be optimized for readability over performance IMO, given it doesn't run very often or for very long